components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -212,18 +256,37 @@ def run_clp(
 
     # Compute the total amount of data compressed
     last_archive_stats = None
-    total_uncompressed_size = 0
-    total_compressed_size = 0
-    while True:
+    last_line_decoded = False
+    worker_output = {
+        "total_uncompressed_size": 0,
+        "total_compressed_size": 0,
+    }
+
+    while not last_line_decoded:
         line = proc.stdout.readline()
-        if not line:
-            break
-        stats = json.loads(line.decode("ascii"))
-        if last_archive_stats is not None and stats["id"] != last_archive_stats["id"]:
+        stats: Optional[Dict[str, Any]] = None
+        if line:
+            stats = json.loads(line.decode("ascii"))
+        else:
+            last_line_decoded = True
+
+        if last_archive_stats is not None and (
+            None is stats or stats["id"] != last_archive_stats["id"]
```

Comment:
not the best way to write the loop, but then we don't have to duplicate everything after the end of the loop.